### PR TITLE
Added form elements, async submission & POST api endpoint

### PR DESCRIPTION
**Pull request by mwyner**
- added form elements to index.html
- added get/post endpoints for user management
- added async form submit script to main.js
- added users.json flat file "database"

@m-ux r?

### CODE REVIEW

**Comment by mcpatten** [comment]
@mwyner :rocket:

**Comment by mcpatten** [comment]
if i add a new comment...

File: dist/index.html [hunk 1]
```diff
@@ -20,7 +20,16 @@
 			<a href="http://www.smashingmagazine.com/2014/06/building-with-gulp/">gulp</a> and
 			<a href="http://expressjs.com/en/starter/static-files.html">express</a>).
 		</p>
+		<form id="create-user">
+			<input class="first-name" type="text" placeholder="Type your first name here">
+			<input class="last-name" type="text" placeholder="Type your last name here">
+			<select class="sex">
```
Comment (mcpatten) [inline dist/index.html:26]: inline comment

Comment (mwyner) [inline dist/index.html:26]: OK, got it

**Comment by mwyner** [comment]
Everything's fine.

**Pull request merged**